pkg/kubelet/images/image_manager.go

```diff
@@ -167,6 +183,13 @@ func (m *imageManager) EnsureImageExists(ctx context.Context, objRef *v1.ObjectR
 		m.backOff.Next(backOffKey, m.backOff.Clock.Now())
 
 		msg, err := evalCRIPullErr(imgRef, imagePullResult.err)
+
+		// Store the actual pull error for providing that information during
+		// the image pull back-off.
+		if !errors.Is(err, ErrImagePull) {
+			m.prevPullErrMsg.Store(backOffKey, fmt.Sprintf("%s: %s", err, msg))
```

Comment:
sorry, I didn't look deep. Is error going before the message is the way we render it? Based on your comment above it looks it is. Just checking

Reply:
Yeah so that users have a chance to parse for the error code itself.

Reply:
Hm, I would say that's intentional and expected. Should we swap it?

Reply:
no, looks ok. I was just surprised we do it in this order